src/services/expenses.service.js

```diff
@@ -0,0 +1,63 @@
+/* eslint-disable function-paren-newline */
+const { Expense } = require('../models/Expense.model');
+const Match = require('../filterExp');
+
+const allExpenses = async (id, categories, from, to) => {
+  const fromDate = from
+    ? new Date(from).toISOString()
+    : new Date(0).toISOString();
+
+  const toDate = to ? new Date(to).toISOString() : new Date().toISOString();
+  const result = await Expense.findAll();
+
+  return result.filter((expense) =>
+    Match.isExpenseMatch(expense, id, categories, fromDate, toDate),
+  );
+};
```

Comment:
You don't need to filter them, just query from the database what you actually need instead of all